### ISSUE TITLE
Inertia is not correctly reported and partial custom inertia is not supported

### ISSUE THREAD

**Issue by ghostsoft** [godot-jolt/godot-jolt#798]
I was looking at the automatically calculated inertia vector and comparing it to the one that the default 3D physics provides and it looks like the y and z are swapped, as if GodotJolt is mistaking Z as the up vector.

For the same object, Default physics gives me this moment of inertia for a large, tall object:
(13.6, 4.05, 13.6)
while GodotJolt gives me
(13.6, 13.6, 4.05)

_However_ I also noticed that manually changing the intertia vector doesn't seem to have a reliable effect on the rigidbody. I've got a body that is very narrow but tall and I wanted to make sure it could spin freely along the y axis but be hard to topple (so in default physics this would be making the x and z inertia very high) but this does not do anything at all in GodotJolt. Changing this to x and y instead of x and z (due to the aforementioned mistake(?) with which direction is up) produces the same result, which is to say it does nothing.

**Comment by mihe** [comment]
> I was looking at the automatically calculated inertia vector and comparing it to the one that the default 3D physics provides and it looks like the y and z are swapped, as if GodotJolt is mistaking Z as the up vector.

I'm not sure how this went unreported for this long, nor how I missed this to begin with, but yes, the inertia as reported by `PhysicsDirectBodyState3D.inverse_inertia` is indeed incorrect. Apparently I had missed that you need to multiply Jolt's `GetInverseInertiaDiagonal()` with its `GetInertiaRotation()` to get the proper local-space inertia. So the inertia is correct within Jolt, but reported incorrectly by this extension.

If you need a workaround until I can publish a fix you should be able to extract the inertia diagonal from the world-space inertia tensor instead, as that one is actually correct, by doing something like this:

```gdscript
var inverse_inertia_tensor := state.transform.basis.transposed() * state.inverse_inertia_tensor * state.transform.basis
var inverse_inertia := Vector3(inverse_inertia_tensor[0][0], inverse_inertia_tensor[1][1], inverse_inertia_tensor[2][2])
```

> I've got a body that is very narrow but tall and I wanted to make sure it could spin freely along the y axis but be hard to topple (so in default physics this would be making the x and z inertia very high) but this does not do anything at all in GodotJolt.

I'm guessing you're leaving the Y-component at zero? It appears I've also missed the fact that Godot Physics allows for a partial custom inertia, where leaving only certain components at zero will still use the automatically computed value for those components. Right now you'll only get the automatically computed value for the inertia if you leave all three components at zero, meaning you just get 0 (which is not valid) if you only leave one or two components at zero. The workaround in this case would be to set a non-zero value for all components if you want to use a custom inertia.

**Comment by mihe** [comment]
The fix for both of these issues can now be found in the latest pre-release, 0.13.0-beta1, which you'll find [here](https://github.com/godot-jolt/godot-jolt/releases).

Note that this version targets the just recently released [Godot 4.3-beta1](https://godotengine.org/article/dev-snapshot-godot-4-3-beta-1/), so won't work with Godot 4.2. Also note that this version is not available through the Godot Asset Library, so you'll need to download it through GitHub.

A proper release will be out some time around Godot 4.3-stable.

**Comment by mihe** [comment]
I figured I'd give a minor update here, as [0.13.0-stable](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.13.0-stable) has just been released, so if you've been relying on 0.13.0-beta1 for this functionality until now I would highly recommend updating, as this most recent release includes some important bug fixes.

Note that the release on the Godot Asset Library is still pending approval as of typing this, so you might need to download it from the link above.
